modules/migrations/base/downloader.go

```diff
@@ -13,9 +13,9 @@ import (
 
 // GetCommentOptions represents an options for get comment
 type GetCommentOptions struct {
-	IssueNumber int64
-	Page        int
-	PageSize    int
+	Context  interface{}
```

Comment:
This feels questionable... I would argue that there has to be something common to all of these forges and these requests. 

At the least a `foreign_issue` must map to a gitea `issue and that mapping is at its simplest:

Gitea index => foreign URL

That mapping should always have at least a function to returna gitea ID and a function to return the foreign id which should be the URL.

Other functions could be done.

But that essentially defines a basic interface that you should be able to rely on here.

Reply:
Agree dont like the change to context interface either!

If we realy need an "context" we should define our own type in base

Reply:
A context is way better than the old `if gitlab` code.
How about:
```
InternalID() int64
ForeignID() int64
```
If the other system shares the issue/pr counter it is enough. If the other system needs more context data a cast to a concrete type is needed.

Reply:
yes just dont use a generic interface if we need one we should define it ourselve